### PR TITLE
solution

### PR DESCRIPTION
**Pull request by mate-academy-dev**

### CODE REVIEW

File: src/rle.js [hunk 1]
```diff
@@ -18,8 +18,27 @@
  */
 
 function rle(source) {
-  // write code here
+  let counter = 0;
+  let result = '';
 
+  for (let i = 0; i < source.length; i++) {
+    let current = source[i];
+    let next = source[i + 1];
```
Comment (roman-mate) [inline src/rle.js]: will this fail if we pass source of single character?

File: src/rle.js [hunk 2]
```diff
@@ -18,8 +18,27 @@
...
+
+    if (current === next) {
+      counter++;
+    } else {
+      result += current;
+
+      if (counter > 0) {
+        result += counter + 1;
```
Comment (roman-mate) [inline src/rle.js]: maybe string interpolation here? It's not obvious that you are concatenating string and int.

File: src/rle.js (hunk 1 above)
Comment (ivadimko) [reply src/rle.js]: @roman-mate thank you for pointing it up. Will add the test-case with a single character.

File: src/rle.js (hunk 2 above)
Comment (ivadimko) [reply src/rle.js]: @roman-mate I've copypasted this solution from one of the students just for test. But thank you, this line is not obvious in general.

**Pull request closed without merging**